### ISSUE TITLE
How is state and events handled with the bridge

### ISSUE THREAD

**Issue by slakbal** [Azure/iotc-device-bridge#16]
Hi,

Would be cool if IoT Central out of the box can handle other network integrations (bi-directional) also. 

We're integration with Thingstream (https://thingstream.io/).

Question, we want to make use of the "events" and  "state" concepts of IoT Central. Does the bridge cater also for these type of messages? 

Thanks!!!

**Comment by peolivei2** [comment]
Hi @slakbal ,

In the future we'll be investigating other solutions for bidirectional integration.

About the second question, device bridge can be used for events and states. The [initial page of the repository](https://github.com/Azure/iotc-device-bridge#instructions) has the following example payload, where `ledColor` is being emitted as a state. Just include it in the `measurements`:

```json
{
    "device": {
        "deviceId": "my-cloud-device"
    },
    "measurements": {
        "temp": 20.31,
        "pressure": 50,
        "humidity": 8.5,
        "ledColor": "blue"
    }
}
```

**Comment by slakbal** [comment]
Aaaahh!! Cool! and Events?

**Comment by peolivei2** [comment]
Correct. Just include the telemetry/state/event in `measurements`, making sure it matches the field name used in the device template.

**Comment by slakbal** [comment]
Awesome schaweeettt!  

Last quick question, is there an REST API available for IoT Central? I can see that an access token can be generated.

Use-case would be to have e.g. a field app with witch the "properties" can be updated for an device? Thus for example with the Consoto demo to build a bit of small tooling to associate an IoT device for example with a specific vending matching and capture the machines data into the "properties"?

**Comment by peolivei2** [comment]
We still don't offer a REST API. The `Access Tokens` are for use with [Azure IoT Central Explorer](https://docs.microsoft.com/en-us/azure/iot-central/howto-use-iotc-explorer).

However, we have a [Microsoft Flow integration](https://docs.microsoft.com/en-us/azure/iot-central/howto-add-microsoft-flow), which you can use to perform actions such as create/update/delete a device when a workflow is triggered.
